cocos/2d/CCParticleSystemQuad.cpp

```diff
@@ -239,11 +239,7 @@ void ParticleSystemQuad::setDisplayFrame(SpriteFrame *spriteFrame)
     CCASSERT(spriteFrame->getOffsetInPixels().isZero(), 
              "QuadParticle only supports SpriteFrames with no offsets");
 
-    // update texture before updating texture rect
-    if ( !_texture || spriteFrame->getTexture()->getName() != _texture->getName())
```

Comment:
Without the fix, here's how particles look like (the entire texture atlas is used)
![wrong](https://cloud.githubusercontent.com/assets/1083134/21220722/ece1e9f2-c2c2-11e6-981f-812a0a668765.jpg)

And with the fix, they look like this (ignore the sprite at bottom left)
![ok](https://cloud.githubusercontent.com/assets/1083134/21220732/f5577a48-c2c2-11e6-8778-a50fc76d05fd.jpg)

By test cases you mean tests in cpp-tests? It's my first commit to cocos2d-x so I don't know my way around... Thanks!